From 4b5e9c2f4aa2c9e8f7c1b2a3d4e5f60718293a4b Mon Sep 17 00:00:00 2001
From: Orange Pi OpenBMC Port <port@example.invalid>
Date: Tue, 25 Aug 2026 17:30:00 +0000
Subject: [PATCH] wireless: fix out-of-tree include paths

The vendor UWE5622 makefiles use $(src) as if it were an absolute path.
With an O= build, Kbuild supplies a relative source directory there, so
the include directory cannot be found and wcn_bus.h is missing.  Resolve
the path from srctree instead.

Upstream-Status: Inappropriate [board-port build fix]
---
 drivers/net/wireless/uwe5622/Makefile           | 2 +-
 drivers/net/wireless/uwe5622/unisocwcn/Makefile | 7 ++++---
 drivers/net/wireless/rtl88x2cs/Makefile          | 2 +-
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/net/wireless/uwe5622/Makefile b/drivers/net/wireless/uwe5622/Makefile
--- a/drivers/net/wireless/uwe5622/Makefile
+++ b/drivers/net/wireless/uwe5622/Makefile
@@ -2,7 +2,7 @@ obj-$(CONFIG_AW_WIFI_DEVICE_UWE5622) += unisocwcn/
 obj-$(CONFIG_WLAN_UWE5622)    += unisocwifi/
 obj-$(CONFIG_TTY_OVERY_SDIO)  += tty-sdio/
 
-UNISOCWCN_DIR := $(shell cd $(src)/unisocwcn/ && /bin/pwd)
+UNISOCWCN_DIR := $(abspath $(srctree)/drivers/net/wireless/uwe5622/unisocwcn)
 UNISOC_BSP_INCLUDE := $(UNISOCWCN_DIR)/include
 export UNISOC_BSP_INCLUDE
 
diff --git a/drivers/net/wireless/uwe5622/unisocwcn/Makefile b/drivers/net/wireless/uwe5622/unisocwcn/Makefile
--- a/drivers/net/wireless/uwe5622/unisocwcn/Makefile
+++ b/drivers/net/wireless/uwe5622/unisocwcn/Makefile
@@ -129,9 +129,10 @@
 ccflags-y += -DCONFIG_WCN_UTILS
 
 #### include path ######
-ccflags-y += -I$(src)/include/
-ccflags-y += -I$(src)/platform/
-ccflags-y += -I$(src)/platform/rf/
+UNISOCWCN_SRC := $(abspath $(srctree)/drivers/net/wireless/uwe5622/unisocwcn)
+ccflags-y += -I$(UNISOCWCN_SRC)/include/
+ccflags-y += -I$(UNISOCWCN_SRC)/platform/
+ccflags-y += -I$(UNISOCWCN_SRC)/platform/rf/
 
 #### add cflag for Customer ######
 ### ---------- Hisilicon start ---------- ###

diff --git a/drivers/net/wireless/rtl88x2cs/Makefile b/drivers/net/wireless/rtl88x2cs/Makefile
--- a/drivers/net/wireless/rtl88x2cs/Makefile
+++ b/drivers/net/wireless/rtl88x2cs/Makefile
@@ -2316,7 +2316,7 @@
 
 ########### HAL_RTL8822C #################################
 ifeq ($(CONFIG_RTL8822C), y)
-include $(TopDIR)/drivers/net/wireless/rtl88x2cs/rtl8822c.mk
+include $(abspath $(srctree)/drivers/net/wireless/rtl88x2cs/rtl8822c.mk)
 EXTRA_CFLAGS += -DCONFIG_RTW_IOCTL_SET_COUNTRY
 endif
 
-- 
2.43.0
